nilearn/datasets/atlas.py

```diff
@@ -1320,11 +1621,18 @@ def fetch_atlas_aal(
                 labels.append(label)
         fdescr = fdescr.replace("SPM 12", version)
 
+    lut = _generate_atlas_look_up_table(
+        "fetch_atlas_aal", index=[int(x) for x in indices], name=labels
+    )
+    _check_look_up_table(lut=lut, atlas=atlas_img)
+
     params = {
         "description": fdescr,
         "maps": atlas_img,
         "labels": labels,
         "indices": indices,
+        "lut": lut,
+        "atlas_type": "deterministic",
```

Comment:
For consistency, I would again suggest to create a variable for atlas_type above or remove the variable created for probabilistic and set it as here for deterministic.